documentation/general/dotnet-run-file.md

```diff
@@ -162,20 +160,24 @@ The generated folders might need to be named differently to avoid clashes with e
 
 The entry-point projects (`Program1` and `Program2` in our example)
 have the shared `.cs` files source-included via `<Content Include="../Shared/**/*.cs" />`.
-We could consider having the projects directly in the top-level folder instead
-but that might result in clashes of build outputs that are not project-scoped, like `project.assets.json`.
-If we did that though, it would be enough to exclude the other entry points rather than including all the shared `.cs` files.
 
-Unless the [artifacts output layout][artifacts-output] is used (which is recommended),
-those implicit projects mean that build artifacts are placed under those implicit directories
-even though they don't exist on disk prior to build:
-```
-App/Program1/bin/
-App/Program1/obj/
-App/Program2/bin/
-App/Program2/obj/
+We could consider using `InternalsVisibleTo` attribute but that might result in slight differences between single- and multi-entry-point programs
+(if not now then perhaps in the future if [some "more internal" accessibility](https://github.com/dotnet/csharplang/issues/6794) is added to C# which doesn't respect `InternalsVisibleTo`)
+which would be undesirable when users start with a single entry point and later add another.
+Also, `InternalsVisibleTo` needs to be added into a C# file as an attribute, or via a complex-looking `AssemblyAttribute` item group into the `.csproj` like:
```

Comment:
Roslyn itself uses items like the following: https://github.com/dotnet/roslyn/blob/4f8f5e6feb80a9c2d651232735ef69d32710927f/src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj#L50

Is that not available to us here?

Reply:
No, I think that's defined by arcade in https://github.com/dotnet/arcade/blob/1741844bd26eb13fc4731b1e9aed218686717fbd/src/Microsoft.DotNet.Arcade.Sdk/tools/GenerateInternalsVisibleTo.targets